### PR TITLE
Fix: duplicate saving and stale webview.

### LINKED ISSUE #511: Switching tabs and back after saving a paste messes the editor.
1. Open a file in a text editor and in the hex editor, so you have two tabs open.
2. Paste some bytes on insert mode in the hex editor and save it.
3. Switch to the text editor and back to the hex editor. 
4. You'll see what you pasted twice. 

bug

Note: Replacing works fine. This only happens when inserting and switching tabs in a way that hides the hex editor.

I attempted to fix this issue but couldn't figure out anything. For some reason, the debugger stops working with all files under ./editor after switching tabs.

Any help would be greatly appreciated.

### PR DESCRIPTION
**Pull request by tomilho**
Should fix #511. Took me a bit of time to figure out everything, but I think the solution is working. I also added some tests to triple check the work.

### CODE REVIEW

File: src/hexDocument.ts [hunk 1]
```diff
@@ -93,12 +93,20 @@ export class HexDocument extends Disposable implements vscode.CustomDocument {
 		return vscode.Uri.parse(this.model.uri);
 	}
 
+	// /**
```
Comment (connor4312) [inline src/hexDocument.ts:96]: Commented code should be removed

**Review by connor4312 — COMMENTED** [review]
Looks good, one comment

**Comment by connor4312** [comment]
Thanks!

**Review by connor4312 — APPROVED** [review]

**Comment by connor4312** [comment]
Hm, it looks like there's a test failure

**Review by connor4312 — APPROVED** [review]

File: src/fileSystemAdaptor.ts [hunk 2]
```diff
@@ -194,9 +194,11 @@ class NativeFileAccessor implements FileAccessor {
 
 	writeBulk(ops: readonly FileWriteOp[]): Promise<void> {
 		return this.handle.borrow<void>(async fd => {
-			for (const { data, offset } of ops) {
-				fd.write(data, 0, data.byteLength, offset);
-			}
+			return Promise.all(
+				ops.map(({ data, offset }) => {
+					return fd.write(data, 0, data.byteLength, offset);
```
Comment (connor4312) [inline src/fileSystemAdaptor.ts:199]: good catch!

**Review by lramos15 — APPROVED** [review]

**Pull request merged**